src/v/kafka/server/handlers/fetch.cc

```diff
@@ -209,17 +226,46 @@ ss::future<read_result> read_from_ntp(
     }
 
     auto high_watermark = partition->high_watermark();
+
     auto max_offset = high_watermark < model::offset(0) ? model::offset(0)
                                                         : high_watermark;
+
+    if (config::shard_local_cfg().enable_transactions.value()) {
```

Comment:
it's easier this way because request context lives on another core and a couple of frames up the call stack, what do you think? I probaly could pass `are_transactions_enabled` bool although

Reply:
ahh, yeh getting at that state is pretty inconvenient. I think its fine to deal with this later.